gramps/plugins/lib/libprogen.py

```diff
@@ -516,32 +515,28 @@ def parse_progen_file(self):
         self.pers = _read_recs(self.def_['Table_1'], self.bname, self.mems)
         self.rels = _read_recs(self.def_['Table_2'], self.bname, self.mems)
 
-        # start feedback about import progress (GUI / TXT)
-        if self.uistate:
-            self.progress.set_pass(_('Initializing'))
-        else:
-            self.reset(_("Import from Pro-Gen"))
-            self.set_total(2 * len(self.pers) + len(self.rels))
+        # calculate total amount of data
+        if not self.uistate:
+            # approx. (1x father, 1x mother) + 1.5x child & families
+            self.set_total(2.5 * len(self.pers) + len(self.rels))
 
+        self.dbase.disable_signals()
         with DbTxn(_("Pro-Gen import"), self.dbase, batch=True) as self.trans:
-            self.dbase.disable_signals()
```

Comment:
self.dbase.disable_signals() is not necessary if you use batch=True; the db does this for you.  (and the matching enable_signals)